File: core/src/main/java/com/github/jackchen/android/core/processor/SampleInterfaceProcessor.java
package com.github.jackchen.android.core.processor;

import androidx.appcompat.app.AppCompatActivity;
import com.github.jackchen.android.core.exception.SampleFailedException;
import com.github.jackchen.android.sample.api.SampleItem;
import com.github.jackchen.android.core.main.sample.SampleInterface;

/**
 * @author Created by cz
 * @date 2020-01-29 22:02
 * @email bingo110@126.com
 */
public abstract class SampleInterfaceProcessor<T> extends ActionProcessor {
    @Override public boolean isAvailable(final Class clazz) {
        return SampleInterface.class.isAssignableFrom(clazz);
    }

    @Override public void execute(final AppCompatActivity context, final SampleItem sampleItem)
            throws SampleFailedException {
        try {
            Class<SampleInterface<T>> clazz = (Class<SampleInterface<T>>) Class.forName(sampleItem.className);
            final SampleInterface<T> sampleInterface = clazz.newInstance();
            final T object = sampleInterface.getObject(context);
            execute(context, object);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public abstract void execute(final AppCompatActivity context, T object) throws SampleFailedException;
}

File: core/src/main/java/com/github/jackchen/android/core/processor/SampleInterfaceProcessor.kt
package com.github.jackchen.android.core.processor

import androidx.appcompat.app.AppCompatActivity
import com.github.jackchen.android.core.exception.SampleFailedException
import com.github.jackchen.android.core.main.sample.SampleInterface
import com.github.jackchen.android.sample.api.SampleItem

/**
 * @author JackChen
 */
abstract class SampleInterfaceProcessor<T> : ActionProcessor {

  override fun isAvailable(clazz: Class<*>): Boolean {
    return SampleInterface::class.java.isAssignableFrom(clazz)
  }

  @Suppress("UNCHECKED_CAST")
  override fun execute(context: AppCompatActivity, sampleItem: SampleItem) {
    try {
      val clazz = sampleItem.clazz() as Class<SampleInterface<T>>
      val sampleInterface: SampleInterface<T> = clazz.newInstance()
      val obj: T = sampleInterface.getObject(context)
      execute(context, obj)
    } catch (e: Exception) {
      e.printStackTrace()
      throw SampleFailedException(sampleItem, e)
    }
  }

  @Throws(SampleFailedException::class)
  abstract fun execute(context: AppCompatActivity, obj: T)
}
